### DISCUSSION TITLE
INITIAL PROPOSAL: DRACO II - Phase 2 - Crypto.org Chain with native token issuance

### DISCUSSION THREAD

**Discussion by lezzokafka** [crypto-org-chain/chain-main#641] [General] [answered]
Dear community members, 

After our internal dry-run was conducted, the [v3.3.2](https://github.com/crypto-org-chain/chain-main/releases/tag/v3.3.2)  `chain-maind` has been released lately.  

We are proposing a network upgrade for  the Crypto.org Chain mainnet - `crypto-org-chain-mainnet-1`. Which implies that validators and full nodes would need to upgrade their `chain-maind` from `v2.*` to `v3.3.2` at some specific time if the network upgrade proposal is passed.

## What is this upgrade about?
**_(Updated 22, Nov, 2021)_**
This version is meant for the [**Draco II - Phase 2**](https://blog.crypto.com/crypto-org-chain-intergalactic-roadmap/) mainnet upgrade to enable native token issuance via IBC solo machines. 

This new binary v3.3.2 is an upgraded base on [Cosmos SDK 0.44.4](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.44.4), [Ibc-go 2.0.0](https://github.com/cosmos/ibc-go/releases/tag/v2.0.0), and [Tendermint v0.34.14](https://github.com/tendermint/tendermint/releases/tag/v0.34.14), which contain a few important updates to the network. 
Specifically, it includes:

- New module [x/authz](https://docs.cosmos.network/master/modules/authz/):   It allows granting arbitrary privileges from one account (the granter) to another account (the grantee) to execute certain messages on its behalf
- New module [x/feegrant](https://docs.cosmos.network/master/modules/feegrant/):  One can authorize allowance for paying for transaction fees from a different account
- New types of public keys and signing algorithms: e.g. `secp256r1`, which is commonly applied in HSMs, iPhone Secure Enclave, will allow these devices to function as hardware wallets basically.
- The vesting account bug was fixed, which solves the previous technical issue on the vesting account and limits the usage of the vesting account. With the fix of this problem, Crossfire winners now can manage the reward in their vesting accounts. 
- Governance “weighted votes”, which allows the validators to split their voting power towards the governance proposal with the power amount they preferred, instead of using all voting power at single voting. 
- The previous binary has improved state-sync related functions so that the interested validators and full node hosts can use it to get state-sync
- Other changes include increasing the security of the IBC module, improving interaction with inactive clients via governance and with relayer, fixing the solo machine light client, etc
- and more... detailed changelog of Cosmos SDK can be found [here](https://github.com/cosmos/cosmos-sdk/blob/v0.44.3/CHANGELOG.md).

We would love to hear from you! You are welcome to leave comments here in this post.

**Comment by eeldivad** [comment]
1.  Will this upgrade require halting the chain or can validators upgrade ASAP but days apart?
2.  Will testnet being upgraded before mainnet?

**Reply by aw126** [reply]
Hello @eeldivad, regarding the questions:
1. As this one is a breaking change, the chain will be paused until 66%+ voting power is back online, just like what we did last time for the NFT upgrade.
2. Testnet `croeseid-4` has already been upgraded to that version.

**Reply by lezzokafka** [reply]
Update: `testnet-croeseid-4` has been updated to the latest version `v3.1.0-croeseid`, which is built with Cosmos SDK to `0.44.0`

**Comment by samuel-tay** [comment]
As this upgrade contains many changes, would like to see stability and stress testing in the testnet before upgrade.

**Reply by aw126** [reply]
Thanks for sharing your thought, yes, we are working on how best to ensure stability for this upgrade, and will keep you guys posted!

**Reply by esspam** [reply]
I would hope that the vesting-account fix is tested as well (don't want to lose the locked and the additional 'free' cros I have in my vesting account 😉)

**Comment by giansalex** [comment, marked as answer]
should update to [cosmos sdk 0.44](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.44.0).

**Reply by aw126** [reply]
@giansalex You are right! we are about to do it. Thanks for the note.

**Reply by lezzokafka** [reply]
Yes, we have been working on that -- see #643

**Reply by lezzokafka** [reply]
Update: [v3.1.0](https://github.com/crypto-org-chain/chain-main/releases/tag/v3.1.0) pre-release is now available.

**Comment by reserg86** [comment]
Greetings, do I correctly understand that a native token creation allowance is proposed? What about CRO?

**Reply by lezzokafka** [reply]
Hello @reserg86, the native token issuance will be performed by [IBC solo machine](https://github.com/crypto-com/ibc-solo-machine). 
It will be leveraging the current IBC token format which allows users to issue tokens natively on Crypto.org Chain. 
Meanwhile the CRO remains as the native token as it is 👍 .

**Reply by reserg86** [reply]
Thank you very much for the clarification!
